lib/fetch/request.js

```diff
@@ -844,6 +844,26 @@ function cloneRequest (request) {
   return newRequest
 }
 
+/**
+ * @param {any} innerRequest
+ * @param {AbortSignal} signal
+ * @param {'request' | 'immutable' | 'request-no-cors' | 'response' | 'none'} guard
+ * @param {any} [realm]
+ * @returns {Request}
+ */
+function fromInnerRequest (innerRequest, signal, guard, realm) {
```

Comment:
Out of curiosity, why use a function for a single place that uses it?
I might be missing the holistic view, but how does this solve the problem?

Reply:
Ok, now I see the usage; resolving my own comment 😅